R/estimate_RIT.R

```diff
@@ -0,0 +1,103 @@
+#' @title estimate RIT score for a student using regression or interpolation
+#' 
+#' @description Given studentid, measurementscale, and a target_date, the 
+#' function will return an estimated score based on selected method 
+#' 
+#' @param mapvizieR_obj mapvizieR object
+#' @param sid target studentid
+#' @param measurescale target subject
+#' @param target_date date of interest, %Y-%m-%d format
+#' @param method which method to use to estimate RIT score
+#' @param foward default is TRUE, set to FALSE if only scores before target_date should be chosen for 'closest' method
+#' 
+#' @export
+
+
+estimate_rit <- function(
+  mapvizieR_obj,
+  sid,
+  measurescale,
+  target_date,
+  method = c('closest','lm','interpolate'),
+  forward = TRUE
+) {
+  
+  # throw an error if method is not given
+  if (missing(method)) {
+    stop('method not given')
+  }
+  
+  # pull out the cdf
+  cdf <- mapvizieR_obj[['cdf']]
+  
+  target_date <- as.Date(target_date)
+  
+  # find the matchings rows
+  if (require(dplyr)) {
```

Comment:
we can assume that `dplyr` is around -- it's a dependency of our package.  that means we can strip out the subset call here